virtcontainers/experimental/README.md

```diff
@@ -0,0 +1,60 @@
+# Experimental package description
+
+## 1. What are "experimental" features?
+
+"Experimental" features are features living in master branch, 
+but Kata community thinks they're not ready for production use.
+They are **always disabled** by default in Kata components releases,
+and can only be enabled by users when they want to have a try.
+
+We suggest you **NEVER** enable "experimental" features in production environment,
+unless you know what breakage they can bring and have confidence to handle it by youself.
+
+Criteria of an experimental feature are:
+
+* the feature breaks backward compatibility
+
+compatibility is important to Kata Containers,
+We will **NEVER** accept any new features/codes which break the backward compatibility of Kata components,
+unless they are so important and there's no way to avoid the breakage.
+If we decide to involve such changes, maintainers will help to make the decision that which Kata release
+it should land.
+
+Before it's landed as a formal feature, we allow the codes to be merged first into our master with the tag "experimental",
+so it can improve in next few releases to be stable enough.
+
+* the feature is not stable enough currently
+
+Some features could be big, it adds/changes lots of codes so may need more tests.
+Our CI can help guarantee correctness of the feature, but it may not cover all scenarios.
+Before we're confident that the feature is ready for production use,
+the feature can be marked as "experimental" first, and users can test it manually in their own environment if intested in it.
+
```

Comment:
I'm not clear if this is supposed to be a list of the criteria we will consider for experimental features, or if it's supposed to be a comprehensive list of possible criteria.

If the latter, we could add:

- the feature increases performance or memory density by at the cost of potentially reducing security somehow.
- the feature is designed for a very specialist / niche scenario but would degrade performance / memory density for the common case.

Reply:
I guess that sounds reasonable. To help "solidify" the whole concept of experimental, I'm trying to think of an existing feature that we *would* have marked as "experimental" if we'd had the ability to do that when the feature was created. So far I haven't found one. Do you have any thoughts on that?

Reply:
I'm not sure, maybe "firecracker" support can be one?